src/components/Toaster/ToastManager/ToastManager.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { memo, useState } from 'react';
+
+import { AlertIntent } from '../../Alert/Alert';
+import Toast from '../Toast/Toast';
+
+import styles from './ToastManager.module.css';
+
+export interface ToastSettings {
+  id?: string;
+  isClosable?: boolean;
+  duration?: number;
+  intent: AlertIntent;
+}
+
+export interface ToastInstance {
+  id: string;
+  message: string;
+  isClosable: boolean;
+  duration: number;
+  onClose: () => void;
+  intent: AlertIntent;
+  isShown: boolean;
+}
+
+export type ToastCallHandler = (message: string, settings: ToastSettings) => void;
+
+interface Props {
+  bindNotify: (handler: ToastCallHandler) => void;
+  bindRemove: (handler: (id: string) => void) => void;
+  bindGetToasts: (handler: () => ToastInstance[]) => void;
+  bindCloseAll: (handler: () => void) => void;
+}
+
+const hasCustomId = (settings: ToastSettings) => Object.hasOwnProperty.call(settings, 'id');
+
+const ToastManager = memo(({ bindNotify, bindRemove, bindGetToasts, bindCloseAll }: Props) => {
+  const [toasts, setToasts] = useState<ToastInstance[]>([]);
+  const [idCounter, setIdCounter] = useState(0);
+
+  const getToasts = () => toasts;
+
+  const closeAll = () => {
+    setToasts(toasts.map((toast) => ({ ...toast, isShown: false })));
+  };
+
+  /**
+   * This will set isShown on the Toast which will close the toast.
+   * It won't remove the toast until onExited triggers onRemove.
+   */
+  const closeToast = (id: string) => {
+    setToasts(
+      toasts.map((toast) => {
+        if (toast.id === id) {
+          return {
+            ...toast,
+            isShown: false,
+          };
+        }
+
+        return toast;
+      }),
+    );
+  };
+
+  const safeCloseToast = (id: string) => {
+    const toastToRemove = toasts.find((toast) => String(toast.id).startsWith(id));
+
+    if (toastToRemove) {
+      closeToast(toastToRemove.id);
+    }
+  };
+
+  const removeToast = (id?: string) => {
+    if (id) {
+      const updatedToasts = toasts.filter((toast) => !String(toast.id).startsWith(id));
+      setToasts(updatedToasts);
+      return updatedToasts;
+    }
+    return toasts;
+  };
+
+  const createToastInstance = (
+    message: string,
+    { isClosable = true, duration = 5, id, intent }: ToastSettings,
+  ): ToastInstance => {
+    const uniqueId = idCounter;
```

Comment:
I think keeping state of the latest id can be avoided when using a Date() timestamp with ms as id